multi_physics/QED/kokkos_example/kokkos_example_quantum_synchrotron.cpp

```diff
@@ -0,0 +1,311 @@
+    Real dt, const int repetitions)
+{
+    Kokkos::Timer timer;
+    for(int rr = 0; rr < repetitions; ++rr){
+        const auto num_particles = pdata.num_particles;
+        Kokkos::parallel_for("EvolveOpt", num_particles, KOKKOS_LAMBDA(int i){
+            const auto px = pdata.m_momentum(i,0);
+            const auto py = pdata.m_momentum(i,1);
+            const auto pz = pdata.m_momentum(i,2);
+            const auto ex = pdata.m_fields.Ex(i);
+            const auto ey = pdata.m_fields.Ey(i);
+            const auto ez = pdata.m_fields.Ez(i);
+            const auto bx = pdata.m_fields.Bx(i);
+            const auto by = pdata.m_fields.By(i);
+            const auto bz = pdata.m_fields.Bz(i);
+            auto& opt = pdata.m_fields.opt(i);
+            const auto ee =
+                pxr::compute_gamma_ele_pos<Real>(px, py, pz)*mec2<Real>;
+            const auto chi =
+                pxr::chi_ele_pos<Real, pxr::unit_system::SI>(
+                    px, py ,pz, ex, ey, ez, bx, by, bz);
+            pxr_qs::evolve_optical_depth<Real, TableType>(
+                ee, chi, dt, opt, ref_table);
+        });
+    }
+
+    Kokkos::fence();
+    const auto time = timer.seconds()*1000.0;
+
+    return std::make_pair(check(pdata.m_fields.opt, true, false), time);
+}
+
+/**
+* Tests photon emission
+*
+* @tparam Real the floating point type to be used
+* @tparam TableType the lookup table type
+* @param[in,out] pdata the particle data
+* @param[in] ref_table the photon emission lookup table
+* @param[in] repetitions how many times should the test be repeated
+* @param[in,out] rand_pool a random pool
+* @return a bool success flag and the elapsed time in ms, packed in a pair
+*/
+template <typename Real, typename TableType>
+std::pair<bool, double>
+generate_photons(
+    ParticleData<Real>& pdata,
+    const TableType& ref_table,
+    const int repetitions,
+    Kokkos::Random_XorShift64_Pool<>& rand_pool)
+{
+    const auto num_particles = pdata.num_particles;
+
+    auto photon_momentum = init_multi_comp_view_with_random_content<Real>(
+        "photon_momentum", 0.0, 0.0, num_particles, rand_pool);
+
+    Kokkos::Timer timer;
+    for(int rr = 0; rr < repetitions; ++rr){
+        Kokkos::parallel_for("PhotEm", num_particles, KOKKOS_LAMBDA(int i){
+            const auto px = pdata.m_momentum(i,0);
+            const auto py = pdata.m_momentum(i,1);
+            const auto pz = pdata.m_momentum(i,2);
+            const auto ex = pdata.m_fields.Ex(i);
+            const auto ey = pdata.m_fields.Ey(i);
+            const auto ez = pdata.m_fields.Ez(i);
+            const auto bx = pdata.m_fields.Bx(i);
+            const auto by = pdata.m_fields.By(i);
+            const auto bz = pdata.m_fields.Bz(i);
+
+            const auto chi = pxr::chi_ele_pos<Real, pxr::unit_system::SI>(
+                px, py ,pz, ex, ey, ez, bx, by, bz);
+
+            auto rand_gen = rand_pool.get_state();
+
+            auto e_phot = pxr_m::vec3<Real>{0,0,0};
```

Comment:
If this is the momentum of the photon, should this variable be called `p_phot` (otherwise the `e` kind of suggests energy)

Reply:
I agree, I am now using a more descriptive name...